src/pubsub.rs

```diff
@@ -0,0 +1,614 @@
+
+        match meta.request_processor.get_signature_status(signature) {
+            Ok(_) => {
+                sink.notify(Ok(RpcSignatureStatus::Confirmed))
+                    .wait()
+                    .unwrap();
+            }
+            Err(_) => {
+                let active_subscriptions = self.subscriptions.clone();
+                thread::spawn(move || {
+                    while active_subscriptions.read().unwrap().contains(&sub_id) {
+                        if meta
+                            .request_processor
+                            .get_signature_status(signature)
+                            .is_ok()
+                        {
+                            sink.notify(Ok(RpcSignatureStatus::Confirmed))
+                                .wait()
+                                .unwrap();
+                            break;
+                        }
+                        sleep(Duration::from_millis(100));
+                    }
+                });
+            }
+        }
+    }
+
+    fn signature_unsubscribe(&self, id: SubscriptionId) -> Result<bool> {
+        let removed = self.subscriptions.write().unwrap().remove(&id);
+        if removed {
+            Ok(true)
+        } else {
+            Err(Error {
+                code: ErrorCode::InvalidParams,
+                message: "Invalid Request: Subscription id does not exist".into(),
+                data: None,
+            })
+        }
+    }
+    fn account_subscribe(
+        &self,
+        meta: Self::Metadata,
+        subscriber: pubsub::Subscriber<Account>,
+        pubkey_str: String,
+    ) {
+        let pubkey_vec = bs58::decode(pubkey_str).into_vec().unwrap();
+        if pubkey_vec.len() != mem::size_of::<Pubkey>() {
+            subscriber
+                .reject(Error {
+                    code: ErrorCode::InvalidParams,
+                    message: "Invalid Request: Invalid pubkey provided".into(),
+                    data: None,
+                }).unwrap();
+            return;
+        }
+        let pubkey = Pubkey::new(&pubkey_vec);
+
+        let id = self.uid.fetch_add(1, atomic::Ordering::SeqCst);
+        let sub_id = SubscriptionId::Number(id as u64);
+        let sink = subscriber.assign_id(sub_id.clone()).unwrap();
+        self.subscriptions.write().unwrap().insert(sub_id.clone());
+
+        let mut previous_userdata = None;
+        let active_subscriptions = self.subscriptions.clone();
+        thread::spawn(move || {
+            while active_subscriptions.read().unwrap().contains(&sub_id) {
+                if let Ok(account) = meta.request_processor.get_account_info(pubkey) {
+                    if Some(account.clone().userdata) != previous_userdata {
+                        sink.notify(Ok(account.clone())).wait().unwrap();
+                        previous_userdata = Some(account.userdata);
+                    }
+                }
+                sleep(Duration::from_millis(100));
```

Comment:
Oh, we definitely want to plug into the bank for account userdata changes somehow.